control/config.py

```diff
@@ -7,40 +7,143 @@
 #  Authors: anita.shekar@ibm.com, sandy.kaur@ibm.com
 #
 
-import configparser
+import os
+import socket
+from pathlib import Path
+from typing import Optional
+from pydantic import (BaseModel, Extra, IPvAnyAddress, PositiveInt, Json,
+                      FilePath, conint, constr, validator, root_validator)
 
 
-class GatewayConfig:
-    """Loads and returns config file settings.
+class GatewaySubConfig(BaseModel):
+    """Validates config file settings and types for gateway.
 
-    Instance attributes:
-        config: Config parser object
+    Required fields report error on missing keys and on invalid values for 
+    present keys.
+
+    Required Fields:
```

Comment:
In some other PR we discussed with @idryomov that it would probably be good to make as much as possible params optional with some reasonable defaults. May be we could do this in this PR?

E.g. I think we could probably have all params optional except `addr` and `port` (`gateway` section), although even for those we could chose defaults like `127.0.0.1` for address and some agreed port number.